shell/common/asar/scoped_temporary_file.cc

```diff
@@ -48,21 +50,28 @@ bool ScopedTemporaryFile::Init(const base::FilePath::StringType& ext) {
   return true;
 }
 
-bool ScopedTemporaryFile::InitFromFile(base::File* src,
-                                       const base::FilePath::StringType& ext,
-                                       uint64_t offset,
-                                       uint64_t size) {
+bool ScopedTemporaryFile::InitFromFile(
+    base::File* src,
+    const base::FilePath::StringType& ext,
+    uint64_t offset,
+    uint64_t size,
+    const absl::optional<IntegrityPayload>& integrity) {
   if (!src->IsValid())
     return false;
 
   if (!Init(ext))
     return false;
 
+  base::ThreadRestrictions::ScopedAllowIO allow_io;
```

Comment:
Surprised this wasn't here before; is it newly needed? We were already doing IO before this PR.

Reply:
This just fixed a DCHECK I hit while testing loading certain files with `fs.createReadStream` 🤷  I assume it would be broken before my changes too